lib/core/src/display/navigation/navigator.rs

```diff
@@ -0,0 +1,158 @@
+use crate::prelude::*;
+
+use super::EventHandler;
+use super::Event;
+use super::ZoomEvent;
+use super::PanEvent;
+use crate::system::web::EventListeningResult as Result;
+use crate::display::rendering::Camera;
+use crate::display::rendering::CameraType;
+use crate::display::rendering::DOMContainer;
+
+use nalgebra::Vector3;
+
+// =================
+// === Animation ===
+// =================
+
+/// Struct used to store animation related properties.
+struct Animation {
+    desired_pos  : Vector3<f32>,
+    vel          : Vector3<f32>,
+    desired_rate : f32,
+    time_count   : f32,
```

Comment:
What does "time count" mean?

Reply:
It's a variable for accumulating the animation time by each frame's `delta_time` (aka `dt`). I renamed it to `accumulated_time`.